### PR TITLE
Scoring updates 

### PR DESCRIPTION
**Pull request by priyakasimbeg**
Addresses:
- https://github.com/mlcommons/algorithmic-efficiency/issues/540
- https://github.com/mlcommons/algorithmic-efficiency/issues/588
- https://github.com/mlcommons/algorithmic-efficiency/issues/589
- https://github.com/mlcommons/algorithmic-efficiency/issues/622
- https://github.com/mlcommons/algorithmic-efficiency/issues/623

Changes to pay attention to during review:
- PRNG Key splitting in run_workloads.py, from submission group to studies to workload trials.
- behavior of `strict` flag in scoring which will put strict requirements regarding number of trials and studies on a submission. 

Todo:
- [x] Add script to generate held-out workloads
- [x] Add held-out workload sampling in script that runs workloads for submission
- [x] Add sampling for studies and trials for in script that runs workloads for submission
- [x] Add instructions for running workloads and scoring submissions

### CODE REVIEW

File: scoring/run_workloads.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+  workloads = [w for w in WORKLOADS.keys()]
+
+  # Read held-out workloads
+  if FLAGS.held_out_workloads_config_path:
+    held_out_workloads = read_held_out_workloads(
+        FLAGS.held_out_workloads_config_path)
+    workloads = workloads + held_out_workloads
+
+  rng_subkeys = prng.split(rng_key, num_studies)[:num_studies:]
```
Comment (priyakasimbeg) [inline scoring/run_workloads.py:138]: double check why index with num_studies

**Review by georgedahl — COMMENTED** [review]
I wrote some of these comments before we met yesterday, so if they contradict that discussion consider them obsolete. Posting them for completeness. Someone else should review the performance profile changes carefully in addition to another full review once you've updated this.

File: scoring/held_out_workloads_example.json
[the remark this answers is not in the document]
Comment (priyakasimbeg) [reply scoring/held_out_workloads_example.json:1]: As discussed offline. The held-out workload config file that is the output of the sampling script only needs the full names of the held out workloads. The scoring code will infer the base workload from the held-out workload name.

File: GETTING_STARTED.md [hunk 2]
```diff
@@ -336,11 +336,44 @@ docker exec -it <container_id> /bin/bash
 ```
 
 ## Score your Submission
+To score your submission we will score over all workloads, held-out workloads and studies as described in the rules. 
+In other words, the total number of runs expected for official scoring is:
+- for external ruleset 8 (workloads) + 6 (held-out workloads) x 5 (studies) x 5
+- for internal ruleset 8 (workloads) + 6 (held-out workloads) x 5 (studies)
 
-To produce performance profile and performance table:
+You may have the time or compute resources to run all required runs, so our scoring scripts will allow some flexibility.
```
[the remark this answers is not in the document]
Comment (priyakasimbeg) [reply GETTING_STARTED.md]: Fixed and moved down to scoring command.

File: scoring/run_workloads.py (hunk 1 above)
Comment (priyakasimbeg) [reply scoring/run_workloads.py]: upon inspection I'm going to conclude that this must have been left-over code from a previous implementation of mine.

**Comment by priyakasimbeg** [comment]
I'm going to merge this into Dev.
Will add tests for code in performance_profile.py in a separate PR in which we can double check the correctness of the scoring calculations.

**Pull request merged**